api/agent/pure_runner.go

```diff
@@ -864,6 +907,31 @@ func (pr *pureRunner) Status(ctx context.Context, _ *empty.Empty) (*runner.Runne
 	return pr.handleStatusCall(ctx)
 }
 
+// BeforeCall called before a function is executed
+func (pr *pureRunner) BeforeCall(ctx context.Context, call *models.Call) error {
+	if call.Type != models.TypeAcksync {
+		return nil
+	}
+	var err error
+	// it is an ack sync we send ResultStart message back
+	pr.callHandleLock.Lock()
+	ch := pr.callHandleMap[call.ID]
+	pr.callHandleLock.Unlock()
+	if ch == nil {
```

Comment:
This can be a panic/logrus.Fatal IMHO. I wouldn't add error codes to try to relay this error.

Reply:
are you sure we want to panic here? I see that we shouldn't be in this scenario, if we can't find a call handle something really bad happened, but is it worth to tear down the service for an acksync error?

Reply:
fixed the bug, thanks